tests/e2e/workloads/bdi/helper.py

```diff
@@ -0,0 +1,143 @@
+        image (str): The name of the image to pull
+        pvc_name (str): The name of the pvc to be used
+        secret_name (str): The secret name to be used
+
+    Returns:
+        dict: The dictionary representing the yaml with all the relevant changes
+    """
+    temp_yaml = templating.load_yaml(constants.IBM_BDI_DATA_LOAD_WORKLOAD_YAML)
+
+    temp_yaml["metadata"]["namespace"] = namespace
+    temp_yaml["spec"]["template"]["spec"]["initContainers"][0]["image"] = image
+    temp_yaml["spec"]["template"]["spec"]["containers"][0]["image"] = image
+    temp_yaml["spec"]["template"]["spec"]["volumes"][0]["persistentVolumeClaim"][
+        "claimName"
+    ] = pvc_name
+    temp_yaml["spec"]["template"]["spec"]["volumes"][1]["secret"][
+        "secretName"
+    ] = secret_name
+
+    return temp_yaml
+
+
+def init_run_workload_yaml(namespace=None, image=None, pvc_name=None, secret_name=None):
+    """Loads the “run-workload-job” and sets all the relevant parameters according to the type of execution
+
+    Args:
+        namespace (str): The name of the namespace
+        image (str): The name of the image to pull
+        pvc_name (str): The name of the pvc to be used
+        secret_name (str): The secret name to be used
+
+    Returns:
+        dict: The dictionary representing the yaml with all the relevant changes
+    """
+    temp_yaml = templating.load_yaml(constants.IBM_BDI_RUN_WORKLOAD_YAML)
+    temp_yaml["metadata"]["namespace"] = namespace
+    temp_yaml["spec"]["template"]["spec"]["initContainers"][0]["image"] = image
+    temp_yaml["spec"]["template"]["spec"]["containers"][0]["image"] = image
+    temp_yaml["spec"]["template"]["spec"]["volumes"][0]["persistentVolumeClaim"][
+        "claimName"
+    ] = pvc_name
+    temp_yaml["spec"]["template"]["spec"]["volumes"][1]["secret"][
+        "secretName"
+    ] = secret_name
+
+    return temp_yaml
+
+
+def install_db2u(class_instance):
+    """Chart installation"""
+
+    os.chdir(class_instance.bdi_dir + class_instance.chart_dir + "/common")
+
+    with open("helm_options", "r") as file:
+        data = file.read()
+        data = data.replace("40Gi", class_instance.pvc_size)
+        with open(class_instance.temp_helm_options.name, "w") as f:
+            f.write(data)
+
+    log.info("Installing db2warehouse")
+    install_cmd = (
+        f"./db2u-install --db-type db2wh --namespace {class_instance.db2u_project} "
+        f"--release-name {class_instance.db2u_r_n} --helm-opt-file "
+        f"{class_instance.temp_helm_options.name} --accept-eula"
+    )
+    exec_cmd(cmd=install_cmd)
+
+    log.info("Waiting for Pods to be in Running/Complete state")
+    pod = ocp.OCP(kind=constants.POD, namespace=class_instance.db2u_project)
+    assert pod.wait_for_resource(
+        condition="Running",
+        selector="app=" + class_instance.db2u_r_n,
+        resource_count=6,
+        timeout=600,
+    )
+
+    assert pod.wait_for_resource(
```

Comment:
is there a way to collect logs from these pods in case of failure analysis? I guess must gather won't collect them.

Reply:
These Pods are being created during db2warehouse installtion. I don't think we need to check their logs.
But, the other Pod's (configure-workload, data-load and run-workload) logs will be collected in the next phase for analysis using nmon